sdk/python/feast/infra/utils/spark_utils.py

```diff
@@ -0,0 +1,24 @@
+from pyspark.sql import DataFrame
+
+from feast.errors import EntitySQLEmptyResults
+
+
+def _assert_empty_sparkDF(df: DataFrame) -> bool:
+
+    """Asserts if a pyspark data frame is empty or not."""
+
+    if df.rdd.isEmpty():
```

Comment:
do you need to check if the df is null?

Reply:
in my POV, it's better practice to check if resulting data frame from executing sql query is empty, without this empty data frame check , user wouldn't know if their sql statement resulted in any data.

This check is optimal, since it's not performing any action, per se.

Users will see an un-related error, much further in execution ( as this entity sql returns NoneType, down stream code breaks with un-related errors as I highlighted in issue: #3324 )

Reply:
ah I see that makes sense